python/tests/test_provenance.py

```diff
@@ -178,6 +220,33 @@ def test_libraries(self):
         assert libs == env["libraries"]
 
 
+class TestGetResources:
+    def test_used_resources_keys(self):
+        resources = provenance.get_resources(_start_time)
+        assert "elapsed_time" in resources
+        assert "user_time" in resources
+        assert "sys_time" in resources
+        if resource is not None:
+            assert "max_memory" in resources
+
+    def test_used_resources_values(self):
+        resources = provenance.get_resources(_start_time)
```

Comment:
Something less fragile would be ``time.time() - delta``, and then test that elapsed time is >= delta.